docs/reference/remap/errors/110_invalid_argument_type.cue

```diff
@@ -0,0 +1,59 @@
+package metadata
+
+remap: errors: "110": {
+	title:       "Invalid argument type"
+	description: """
+		An argument passed to a [function call expression](\(urls.vrl_expressions)#\(remap.literals.regular_expression.anchor))
+		is not a supported type.
+		"""
+	rationale:   """
+		VRL is [type-safe](\(urls.vrl_type_safety)) and requires that types align upon compilation. This contributes
+		heavily to VRL's [safety principle](\(urls.vrl_safety)), ensuring that VRL programs are reliable once deployed.
+		"""
+	resolution: #"""
+		You must guarantee the type of the variable by using the appropriate [type](\(urls.vrl_functions)#type) or
+		[coercion](\(urls.vrl_functions)#coerce) function.
+		"""#
+
+	examples: [...{
+		source: #"""
+			downcase(.message)
+			"""#
+		raises: compiletime: #"""
+			error: \#(title)
+			  ┌─ :1:1
+			  │
+			1 │ downcase(.message)
+			  │          ^^^^^^^^
+			  │          │
+			  │          this expression resolves to unknown type
+			  |          but the parameter "value" expects the exact type "string"
+			  │
+			"""#
+	}]
+
+	examples: [
+		{
+			"title": "\(title) (guard with defaults)"
+			diff: #"""
+				+.message = string(.message) ?? ""
+				 downcase(.message)
+				"""#
+		},
+		{
+			"title": "\(title) (guard with errors)"
+			diff: #"""
+				+string!(.message)
+				 downcase(.message)
```

Comment:
This doesn't work, because you still have to assign the result of the `string` call to have the compiler deduce the type:

```rust
.message = string!(.message)
downcase(.message)
```

or:

```rust
downcase(string!(.message))
```

While technically the current example could be made to work, I'm hesitant to do so, as it would mean one of two things:

* we'd have to special-case `string` (etc) functions (basically converting them to a function-like built-in) to allow manipulating compiler type state, or
* allow functions in general to manipulate type state, which I suspect would be a recipe for breaking core language safety guarantees by loading a badly implemented function.